### ISSUE TITLE
Error when attempting to compose an OSv package. Get signal: abort trap error

### ISSUE THREAD

**Issue by techmonte** [cloudius-systems/capstan#185]
Running into an issue with the Capstan tool when attempting to compose a package using the `capstan package compose...` command. Upon executing the command, the composition process begins and after executing the `Setting:cmdline: --norandom --nomount` task, the process immediately halts with a `signal: abort trap` error. I've been unable to locate anything within the tutorial or help documentation that assists in troubleshooting this error.

**Comment by wkozaczuk** [comment]
Hi,

Can you give us more specifics as to which version of capstan you are
using, on what host, full capstan compose command, and version of OSv
kernel?

Waldek

**Comment by techmonte** [comment]
Using the capstan binary found at https://github.com/cloudius-systems/capstan/releases/tag/v0.5.0.  When running `capstan --version`, I get 0.0.0.  Running on a OSX host (Catalina 10.15.7). Compose command is as folllows: `capstan package compose --size 200MB com.example.word-finder`. Was under the impression that the capstan tool uses a pre-build kernel?

**Comment by techmonte** [comment]
Just following up on my last message.  Does this provide the information needed?

**Comment by wkozaczuk** [comment]
Can you also run the capstan command with the verbose flag and send us its
output? Before it, can you also delete $HOME/.capstan directory?

BTW. There was something wrong with the process of generating the OSX
version of the capstan binary. It should be 0.5.0 and I will fix it.

**Comment by techmonte** [comment]
Verbose flag doesn't show up in the list of available commands/options.  Can you provide the syntax?

**Comment by wkozaczuk** [comment]
```
capstan package compose --help
NAME:
   capstan package compose - composes the package and all its dependencies into OSv image

USAGE:
   capstan package compose [command options] image-name

OPTIONS:
   --size value          total size of the target image (use M or G suffix) (default: "10G")
   --update              updates the existing target VM by uploading only modified files (default: false)
   --verbose             verbose mode (default: false)
   --run value           the command line to be executed in the VM
   --pull-missing        attempt to pull packages missing from a local repository (default: false)
   --boot value          specify default config_set name to boot unikernel with (repeatable, will be run left to right)
   --env value           specify value of environment variable e.g. PORT=8000 (repeatable)
   --fs value            specify type of filesystem: zfs or rofs
   --require value       specify extra package dependency
   --loader_image value  the base loader image (default: "osv-loader")
   --help, -h            show help (default: false)
```

**Comment by techmonte** [comment]
`Command line will be set based on config_set_default attribute of meta/run.yaml`
`Resolved runtime into: node`
`Prepending 'node' runtime dependencies to dep list: [node-4.4.5]`
`extractPackageContent: osv.bootstrap`
`extractPackageContent: node-4.4.5`
`WARN: .capstanignore not found, all files will be uploaded`
`.capstanignore: ignore /meta/package.yaml`
`.capstanignore: ignore /mpm-pkg (entire folder)`
`Failed to download default loader image (osv-loader).`
`Loader image could not be found or downloaded.`

`Failed to initialize empty image named com.example.word-finder.`
`Error was: The request https://api.github.com/repos/cloudius-systems/osv/releases returned non-200 [403] response:` `{"message":"API rate limit exceeded for 72.21.196.64. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://developer.github.com/v3/#rate-limiting"}`

**Comment by techmonte** [comment]
Note that when trying to manually pull mike/osv-loader, I receive a `gzip: invalid header` error

**Comment by wkozaczuk** [comment]
It seems you may possibly have a wrong/old (mikelangelo?) version of capstan.

This works for me:
```
~/Downloads/darwin_capstan package compose --pull-missing --require osv.node-js node-example
```

And then:
```
capstan run -p qemu -v --execute '/libnode.so' node-example   
Removing orphaned instance folder: /Users/wkozaczuk/.capstan/instances/qemu/node-example
Command line will be set based on --run parameter
Created instance: node-example
Setting cmdline: /libnode.so
Running QEMU without acceleration: please install Intel HAXM from https://github.com/intel/haxm/releases
Invoking QEMU at: /usr/local/bin/qemu-system-x86_64 with arguments:
  -vnc :1
  -m 1024
  -smp 2
  -device virtio-blk-pci,id=blk0,bootindex=0,drive=hd0
  -drive file=/Users/wkozaczuk/.capstan/instances/qemu/node-example/disk.qcow2,if=none,id=hd0,aio=threads,cache=unsafe
  -chardev stdio,mux=on,id=stdio,signal=off
  -device isa-serial,chardev=stdio
  -netdev user,id=un0,net=192.168.122.0/24,host=192.168.122.1
  -device virtio-net-pci,netdev=un0
  -chardev socket,id=charmonitor,path=/Users/wkozaczuk/.capstan/instances/qemu/node-example/osv.monitor,server,nowait
  -mon chardev=charmonitor,id=monitor,mode=control
OSv v0.55.0-2-gdbfb0bb1
eth0: 192.168.122.15
Booted up in 0.00 ms
Cmdline: /libnode.so
> console.log(process.version);
v8.11.2
undefined
> 
```

Did you download capstan from here - https://github.com/cloudius-systems/capstan/releases/tag/v0.5.0?

**Comment by wkozaczuk** [comment]
The capstan downloaded from there actually show correct version:
```
~/Downloads/darwin_capstan --version                                                        
capstan version v0.5.0
```

You can obviously rename it to just `capstan` and add it to your path.

**Comment by wkozaczuk** [comment]
BTW this readme - https://github.com/cloudius-systems/capstan#capstan - along with many links in it should provide you with some documentation. It should be fairly up to date.

**Comment by techmonte** [comment]
Verified I'm running v0.5.0 of capstan. Tried a different sample (Java App) and still arrive to the same `signal: abort trap` error when trying to compose the package. Output below:

`capstan package compose java-example`
`Command line will be set to default boot`
`extractPackageContent: osv.bootstrap`
`extractPackageContent: osv.run-java`
`extractPackageContent: osv.openjdk10-java-base`
`extractPackageContent: osv.run-java`
`Importing java-example...`
`Importing into /Users/lamonhar/.capstan/repository/java-example/java-example.qemu`
`Uploading files to /Users/lamonhar/.capstan/repository/java-example/java-example.qemu...`
`Setting cmdline: --norandom --nomount --noinit /tools/mkfs.so; /tools/cpiod.so --prefix /zfs/zfs; /zfs.so set` `compression=off osv`
`signal: abort trap`

BTW, tried running the command you mentioned earlier `capstan run -p qemu -v --execute '/libnode.so' node-example`, but capstan complains that node-example doesn't exist.  This assumes that I've already pulled a package called node-example.

**Comment by techmonte** [comment]
Is there a recorded video that walks through example usage of capstan on macOS?  I need to demo OSv as part of an operating systems grad course I'm taking and can go with a "plan B" if I can't get this running in time.

**Comment by wkozaczuk** [comment]
Did you remove /Users/lamonhar/.capstan directory as I asked you?

And then run my commands. It should re-download everything from github. If
it worked on my OSX Catalina it should work for you as well.

BTW the 'node-example' is not a package name. It is an image name created
by the package compose command. Which QEMU version do you have?

I have 5.1.0

**Comment by techmonte** [comment]
Yes, removed .capstan directory.  Running version 5.1.0 of qemu.  Get the following output when running the command you referenced.

`capstan run -p qemu -v --execute '/libnode.so' node-example`
`Command line will be set based on --run parameter`
`node-example: no such image at: /Users/lamonhar/.capstan/repository/node-example/node-example.qemu`

**Comment by wkozaczuk** [comment]
Did you run this command before?
```
capstan package compose --pull-missing --require osv.node-js node-example
```

This one creates the image **node-example** that the other command will try to run

**Comment by wkozaczuk** [comment]
1. `rm -rf /Users/lamonhar/.capstan`
2. `capstan package compose --pull-missing --require osv.node-js node-example`
3. `capstan run -p qemu -v --execute '/libnode.so' node-example`

**Comment by techmonte** [comment]
Yeah I did.  Running into GitHub API rate limiting.

**Comment by techmonte** [comment]
Still getting the `signal: abort trap` error.  Output below:

...
`Downloaded loader image (osv-loader) from github.`
`Importing** node-example...`
`Importing into /Users/lamonhar/.capstan/repository/node-example/node-example.qemu`
`Uploading files to /Users/lamonhar/.capstan/repository/node-example/node-example.qemu...`
`Setting cmdline: --norandom --nomount --noinit /tools/mkfs.so; /tools/cpiod.so --prefix /zfs/zfs; /zfs.so set` `compression=off osv`
`signal: abort trap`

**Comment by wkozaczuk** [comment]
I am not sure how to help you with this issue. I am not experiencing the GitHub API rate limiting.. You can manually download the packages and place them in the correct places:

```
~/Downloads/darwin_capstan package compose --verbose --pull-missing --require osv.node-js node-example
Command line will be set to default boot
Manifest file /Users/wkozaczuk/projects/osv-examples/node/meta/package.yaml does not exist. Assuming default manifest
Downloading osv.bootstrap.yaml... from https://github.com/cloudius-systems/osv/releases/download/v0.55.0/osv.bootstrap.yaml
135 B / 135 B [--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% ? p/s
Downloading osv.bootstrap.mpm... from https://github.com/cloudius-systems/osv/releases/download/v0.55.0/osv.bootstrap.mpm
308.48 KiB / 308.48 KiB [--------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 19.41 MiB p/s
Downloading osv.node-js.yaml... from https://github.com/cloudius-systems/osv/releases/download/v0.51.0/osv.node-js.yaml
103 B / 103 B [--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% ? p/s
Downloading osv.node-js.mpm... from https://github.com/cloudius-systems/osv/releases/download/v0.51.0/osv.node-js.mpm
10.06 MiB / 10.06 MiB [---------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 614.69 KiB p/s
extractPackageContent: osv.bootstrap
extractPackageContent: osv.node-js
WARN: .capstanignore not found, all files will be uploaded
.capstanignore: ignore /mpm-pkg (entire folder)
Downloading osv-loader/osv-loader.qemu... from https://github.com/cloudius-systems/osv/releases/download/v0.55.0/osv-loader.qemu
4.47 MiB / 4.47 MiB [-----------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% 711.78 KiB p/s
Downloading osv-loader/index.yaml... from https://github.com/cloudius-systems/osv/releases/download/v0.55.0/index.yaml
115 B / 115 B [--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% ? p/s
Downloaded loader image (osv-loader) from github.
Importing node-example...
Importing into /Users/wkozaczuk/.capstan/repository/node-example/node-example.qemu
Uploading files to /Users/wkozaczuk/.capstan/repository/node-example/node-example.qemu...
Setting cmdline: --norandom --nomount --noinit /tools/mkfs.so; /tools/cpiod.so --prefix /zfs/zfs; /zfs.so set compression=off osv
Invoking QEMU at: /usr/local/bin/qemu-system-x86_64 with arguments:
  -vnc :1
  -m 512
  -smp 0
  -device virtio-blk-pci,id=blk0,bootindex=0,drive=hd0
  -drive file=/Users/wkozaczuk/.capstan/repository/node-example/node-example.qemu,if=none,id=hd0,aio=threads,cache=unsafe
  -chardev stdio,mux=on,id=stdio,signal=off
  -device isa-serial,chardev=stdio
  -netdev user,id=un0,net=192.168.122.0/24,host=192.168.122.1,hostfwd=tcp::10000-:10000
  -device virtio-net-pci,netdev=un0
  -chardev socket,id=charmonitor,path=,server,nowait
  -mon chardev=charmonitor,id=monitor,mode=control
  -accel hax
HAX is working and emulator runs in fast virt mode.
OSv v0.55.0-2-gdbfb0bb1
eth0: 192.168.122.15
Booted up in 0.00 ms
Cmdline: /tools/mkfs.so; /tools/cpiod.so --prefix /zfs/zfs; /zfs.so set compression=off osv
Running mkfs...
Waiting for connection from host...
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/libenviron.so  --> /libenviron.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/run  --> /run 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/run.sh  --> /run.sh 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/tools/mount-fs.so  --> /tools/mount-fs.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/usr  --> /usr 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/usr/lib  --> /usr/lib 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/dev  --> /dev 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/etc/mnttab  --> /etc/mnttab 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/libnode.so  --> /libnode.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/proc  --> /proc 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/tools/cpiod.so  --> /tools/cpiod.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/tools/umount.so  --> /tools/umount.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/zpool.so  --> /zpool.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/etc/hosts  --> /etc/hosts 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/libzfs.so  --> /libzfs.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/tmp  --> /tmp 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/tools  --> /tools 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/usr/lib/libgcc_s.so.1  --> /usr/lib/libgcc_s.so.1 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/zfs.so  --> /zfs.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/etc  --> /etc 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/etc/fstab  --> /etc/fstab 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/libuutil.so  --> /libuutil.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/libvdso.so  --> /libvdso.so 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/sys  --> /sys 
Adding /Users/wkozaczuk/projects/osv-examples/node/mpm-pkg/tools/mkfs.so  --> /tools/mkfs.so 
Command line set to: 'runscript /run/default;'
```

The `.capstan` directory layout:
```
find ~/.capstan 
/Users/wkozaczuk/.capstan
/Users/wkozaczuk/.capstan/repository
/Users/wkozaczuk/.capstan/repository/osv-loader
/Users/wkozaczuk/.capstan/repository/osv-loader/osv-loader.qemu
/Users/wkozaczuk/.capstan/repository/osv-loader/index.yaml
/Users/wkozaczuk/.capstan/repository/node-example
/Users/wkozaczuk/.capstan/repository/node-example/node-example.qemu
/Users/wkozaczuk/.capstan/repository/node-example/node-example.qemu.cache
/Users/wkozaczuk/.capstan/repository/node-example/index.yaml
/Users/wkozaczuk/.capstan/packages
/Users/wkozaczuk/.capstan/packages/osv.node-js.mpm
/Users/wkozaczuk/.capstan/packages/osv.node-js.yaml
/Users/wkozaczuk/.capstan/packages/osv.bootstrap.yaml
/Users/wkozaczuk/.capstan/packages/osv.bootstrap.mpm
```

I am not sure where the `signal: abort trap` comes from.

What is your QEMU version?

Also, try to build an image with Read-Only FS:
```
capstan package compose --verbose --pull-missing --require osv.node-js --fs rofs node-example
```

```
_capstan run -p qemu -v --execute '/libnode.so' node-example                                
Command line will be set based on --run parameter
Created instance: node-example
Setting cmdline: /libnode.so
Invoking QEMU at: /usr/local/bin/qemu-system-x86_64 with arguments:
  -vnc :1
  -m 1024
  -smp 2
  -device virtio-blk-pci,id=blk0,bootindex=0,drive=hd0
  -drive file=/Users/wkozaczuk/.capstan/instances/qemu/node-example/disk.qcow2,if=none,id=hd0,aio=threads,cache=unsafe
  -chardev stdio,mux=on,id=stdio,signal=off
  -device isa-serial,chardev=stdio
  -netdev user,id=un0,net=192.168.122.0/24,host=192.168.122.1
  -device virtio-net-pci,netdev=un0
  -chardev socket,id=charmonitor,path=/Users/wkozaczuk/.capstan/instances/qemu/node-example/osv.monitor,server,nowait
  -mon chardev=charmonitor,id=monitor,mode=control
  -accel hax
HAX is working and emulator runs in fast virt mode.
OSv v0.55.0-2-gdbfb0bb1
eth0: 192.168.122.15
Booted up in 0.00 ms
Cmdline: /libnode.so
> 
Error: Could not open history file.
REPL session history will not be persisted.
> 
(To exit, press ^C again or type .exit)
> 
```

**Comment by wkozaczuk** [comment]
I wonder if that error comes from the OSX host when running capstan. Maybe the capstan binary does not work for some reason on your version of catalina.

Can you try to rebuild capstan from sources on your Mac - https://github.com/cloudius-systems/capstan/wiki/Capstan-Installation?

**Comment by techmonte** [comment]
Got past the rate limiting issue.  Rate limiting comes into play when you exceed the number of API requests coming from a single IP.  Just need to renew your IP and that addresses it. I'm using qemu version 5.1.0.  The issue is the `signal: abort trap` error.  I was able to successfully build the image with Read-Only FS, but when running the subsequent command `capstan run -p qemu -v --execute '/libnode.so' node-example`, the `signal: abort trap` error happens.

**Comment by wkozaczuk** [comment]
Can you bypass capstan when running OSv image and boot OSv directly using qemu like so:

```
/usr/local/bin/qemu-system-x86_64 \
  -vnc :1 \
  -m 1024 \
  -smp 2 \
  -device virtio-blk-pci,id=blk0,bootindex=0,drive=hd0 \
  -drive file=/Users/wkozaczuk/.capstan/instances/qemu/node-example/disk.qcow2,if=none,id=hd0,aio=threads,cache=unsafe \
  -chardev stdio,mux=on,id=stdio,signal=off \
  -device isa-serial,chardev=stdio \
  -netdev user,id=un0,net=192.168.122.0/24,host=192.168.122.1 \
  -device virtio-net-pci,netdev=un0 \
  -chardev socket,id=charmonitor,path=/Users/wkozaczuk/.capstan/instances/qemu/node-example/osv.monitor,server,nowait \
  -mon chardev=charmonitor,id=monitor,mode=control
```

BTW if you try to run OSv image using capstan with `-v` flag it should show you your full QEMU command line.

What is the full output to the screen when you run OSv using capstan and get this `abort` error?

**Comment by wkozaczuk** [comment]
My suspicion is that for whatever reason QEMU fails to launch and does not even get to the point OSv boots.

**Comment by techmonte** [comment]
When attempting to bypass capstan, I get an error that liblzo2.2.dylib library is not loaded (Reason: image not found).  The full  output of running the command doesn't yield what you show in your output.  To your point, the QEMU output doesn't even show (even though I have QEMU installed and running the latest version).  See output below:

`capstan run -p qemu -v --execute '/libnode.so' node-example`
`Removing orphaned instance folder: /Users/lamonhar/.capstan/instances/qemu/node-example`
`Command line will be set based on --run parameter`
`Created instance: node-example`
`Setting cmdline: /libnode.so`
--THERE SHOULD BE SOME QEMU OUTPUT HERE--
`signal: abort trap`

**Comment by wkozaczuk** [comment]
How did you install QEMU? Using brew? That is what I did I think. Can you try to re-install it?

In case you are stuck here is an old demo using old capstan - https://www.youtube.com/watch?v=kAuBPrYk354&feature=youtu.be.

**Comment by techmonte** [comment]
Reinstalled Qemu.  Different error now.  Output below:

`capstan package compose --pull-missing --require osv.node-js node-example`
`Command line will be set based on config_set_default attribute of meta/run.yaml`
`Resolved runtime into: node`
`Prepending 'node' runtime dependencies to dep list: [node-4.4.5]`
`extractPackageContent: osv.bootstrap`
`extractPackageContent: node-4.4.5`
`extractPackageContent: osv.node-js`
`Converting image /var/folders/lm/17zw_cx50k30tx278pq0bwjmfh4zzj/T/capstan889043349/application.img to QCOW2` `format failed in qemu-img`
`Failed to initialize empty image named node-example.`
`Error was: exec: "qemu-img": executable file not found in $PATH`

also using --fs rofs option:

`capstan package compose --pull-missing --require osv.node-js --fs rofs node-example`
`Command line will be set based on config_set_default attribute of meta/run.yaml`
`Resolved runtime into: node`
`Prepending 'node' runtime dependencies to dep list: [node-4.4.5]`
`extractPackageContent: osv.bootstrap`
`extractPackageContent: node-4.4.5`
`extractPackageContent: osv.node-js`
`Resizing /var/folders/lm/17zw_cx50k30tx278pq0bwjmfh4zzj/T/capstan719655329/application.img to new size 201378512b failed in qemu-img`
`Failed to set the target size (201378512b) of the image` `/var/folders/lm/17zw_cx50k30tx278pq0bwjmfh4zzj/T/capstan719655329/application.img`
`Failed to create ROFS image named node-example.`
`Error was: exec: "qemu-img": executable file not found in $PATH`

**Comment by wkozaczuk** [comment]
I think `qemu-img` is part of QEMU. I have it on my system. Are you sure
where qemu is installed in your PATH?

**Comment by techmonte** [comment]
Added QEMU to my PATH (assumed the brew installation would have done that. bad assumption :-)). QEMU image seems to build successful.  However, when executing the capstan run command, still back to the `signal trap error`.  QEMU doesn't seem to be involved. Same output:

QEMU image built...

`capstan package compose --pull-missing --require osv.node-js --fs rofs node-example`
`Command line will be set based on config_set_default attribute of meta/run.yaml`
`Resolved runtime into: node`
`Prepending 'node' runtime dependencies to dep list: [node-4.4.5]`
`extractPackageContent: osv.bootstrap`
`extractPackageContent: node-4.4.5`
`extractPackageContent: osv.node-js`
`Importing node-example...`
`Importing into /Users/lamonhar/.capstan/repository/node-example/node-example.qemu`
`Command line set to: 'runscript /run/word_count;'`

and then...

`capstan run -p qemu -v --execute '/libnode.so' node-example`
`Removing orphaned instance folder: /Users/lamonhar/.capstan/instances/qemu/node-example`
`Command line will be set based on --run parameter`
`Created instance: node-example`
`Setting cmdline: /libnode.so`
--QEMU should be invoked here--
`signal: abort trap`

**Comment by techmonte** [comment]
It appears QEMU is looking for a library that doesn't exist.  It's looking for `liblzo2.2.dylib` that I believe is available on El Capitan but not the Catalina build of MacOS

**Comment by techmonte** [comment]
Made progress @wkozaczuk.  Resolved QEMU issues for the most part with one exception.  Upon trying to compose a package, I get the QEMU error `qemu: could not load PC BIOS 'bios-256k.bin` Is this an old seabios binary or perhaps something else?  If it's the former, is there something I need to install in addition to QEMU?

**Comment by wkozaczuk** [comment]
This bios part is part of QEMU and should be installed together. I very
rarely use QEMU on Mac so I would try to google about QEMU installation
problems on Catalina. BTW I have Catalina as well.

**Comment by wkozaczuk** [comment]
Something is still messed up with your QEMU installation. I think there is
‘-bios’ option which you can use to point to this file provided you locate
it in your filesystem.

**Comment by techmonte** [comment]
Where would you place the '-bios' option?  I've already located where the files are located.  Just need to somehow point QEMU to that location.  Does capstan assist with this?

**Comment by wkozaczuk** [comment]
This is QEMU option and normally you do not need to set it. This indicates
something h wrong with QEMU installation. Maybe wrong PATH or
LD_LIBrARy_PATH.

**Comment by techmonte** [comment]
I think I've fixed it.  I completely uninstalled/reinstalled QEMU.  There was certainly something up with how Homebrew installed QEMU, and when I did a reinstall, I did so on the bad QEMU installation.  Upon completely uninstalling and then reinstalling QEMU, that appeared to address the issue.  I was able to successfully get the Node.js 'word finder' sample built and running.  

The root cause of the `signal: trap` error appears to be a bad QEMU installation and the fact that it was performing an operation that caused a fault in OSX. There was no additional logs generated that helped identify this root cause and really its discovery was a result of trial and error.  It's important to you test out the QEMU installation before proceeding with the rest of the steps cited in the capstan installation instructions.  Otherwise, it will haunt you later when trying to build/run images.  Even with verbose logging turned on in capstan, there was nothing captured with respect to the issues with QEMU that could have helped get to a quicker solution.

There are a lot of nuances in getting capstan running that aren't clearly spelled out in the documentation.  I think assumptions are made (and understandably so) that running the recommended commands yield the desired outcome, but that isn't always the case and there's very limited troubleshooting documentation that exists that helps identify root causes of issues.  

Also note that that a number of OSv samples assumes existence of older packages that no longer exist in the cloudius github repository. For example, the capstan-example-java sample assumes the existence of the osv-openjdk repo (e.g. https://github.com/cloudius/osv-openjdk) which doesn't exist, so mileage may vary depending on what samples you choose to test out. 

Hopefully this thread will be beneficial for anyone else who runs into this particular issue.  

Thanks @wkozaczuk for your support.  It was very much appreciated.
